### PR TITLE
Update blueprints to use native JS classes

### PR DESCRIPTION
**Pull request by ppcano**
Implements `Update blueprints for each object type to use native JS classes` of https://github.com/emberjs/ember.js/issues/17234

Related PRs https://github.com/ember-cli/ember-octane-blueprint/pull/36 and https://github.com/ember-cli/ember-octane-blueprint/issues/37

### CODE REVIEW

File: node-tests/blueprints/controller-test.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const expect = chai.expect;
 
 const generateFakePackageManifest = require('../helpers/generate-fake-package-manifest');
 const enableModuleUnification = require('../helpers/module-unification').enableModuleUnification;
+const enableOctane = require('../helpers/setup-test-environment').enableOctane;
```
Comment (ppcano) [inline node-tests/blueprints/controller-test.js]: I recommend to include the `enableModuleUnification ` function into the `/helpers/setup-test-environment` module, but I would like to do that in an independent PR.

File: blueprints/controller/index.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 'use strict';
 
+const useNativeDetector = require('../native-detector');
```
Comment (rwjblue) [inline blueprints/controller/index.js]: Could we perhaps call this "edition-detector"?

File: blueprints/native-detector.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const path = require('path');
+
+module.exports = function(blueprint) {
+  blueprint.filesPath = function() {
+    let rootPath = process.env.EMBER_VERSION === 'OCTANE' ? 'native-files' : 'files';
```
Comment (rwjblue) [inline blueprints/native-detector.js]: For reference, we intend to come up with a better "flag" than `process.env.EMBER_VERSION`, but this unlocks actually being able to do the blueprint work while we figure that out.

**Review by rwjblue — COMMENTED** [review]
Only one minor suggested tweak.

**Review by rwjblue — APPROVED** [review]
LGTM! Thanks for working on this!

File: blueprints/controller/index.js (hunk 2 above)
Comment (ppcano) [reply blueprints/controller/index.js]: Resolved by https://github.com/emberjs/ember.js/pull/17621/commits/698e8b5325ca63334367e7befb93f2224cf51714

**Comment by ppcano** [comment]
@rwjblue, I have included the other blueprints and this is ready for merging now.   

Do we have also to support Native classes in the Ember Data blueprints? `ember-decorators` includes them, but they are not described in the Octane tracking issue.  

https://github.com/ember-cli/ember-octane-blueprint/issues/37 is related to this PR in case you want to share your opinion on how/if to get rid of the `ember-decorators` blueprints.

**Review by rwjblue — APPROVED** [review]

File: blueprints/edition-detector.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const path = require('path');
+
+module.exports = function(blueprint) {
+  blueprint.filesPath = function() {
+    let rootPath = process.env.EMBER_VERSION === 'OCTANE' ? 'native-files' : 'files';
```
Comment (NullVoxPopuli) [inline blueprints/edition-detector.js]: Wasn't this edition in the blueprint? Am I misremembering?

Comment (ppcano) [reply blueprints/edition-detector.js]: I don't understand the question.

Here is checking if the `ENV.EMBER_VERSION` is equal to 'OCTANE` to use Native files instead of Ember `extend`.

https://github.com/ember-cli/ember-octane-blueprint/pull/36 set the env variable.

Comment (NullVoxPopuli) [reply blueprints/edition-detector.js]: ah ok, you right. I was misremembering. thank you! :)
